lisp/plugins/action_cues/collection_cue.py

```diff
@@ -42,24 +45,50 @@
 class CollectionCue(Cue):
     Name = QT_TRANSLATE_NOOP("CueName", "Collection Cue")
     Category = QT_TRANSLATE_NOOP("CueCategory", "Action cues")
+    CueActions = (CueAction.Default, CueAction.Start, CueAction.Stop)
 
     targets = Property(default=[])
+    tracking = Property(default=False)
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.name = translate("CueName", self.Name)
+        self.ended_cues_count = 0
 
     def __start__(self, fade=False):
         for target_id, action in self.targets:
             cue = self.app.cue_model.get(target_id)
             if cue is not self:
+                if self.tracking:
+                    cue.stopped.connect(self.mark_as_done)
+                    cue.error.connect(self.mark_as_done)
+                    cue.interrupted.connect(self.mark_as_done)
+                    cue.end.connect(self.mark_as_done)
                 cue.execute(action=CueAction[action])
 
-        return False
+        return self.tracking
 
+    def __stop__(self, fade=False):
+        for target_id, action in self.targets:
+            target_cue = self.app.cue_model.get(target_id)
+            target_cue.end.disconnect(self.mark_as_done)
+            target_cue.stopped.disconnect(self.mark_as_done)
+            target_cue.interrupted.disconnect(self.mark_as_done)
+            target_cue.error.disconnect(self.mark_as_done)
+            target_cue.stop()
+        self.ended_cues_count = 0
+
+        return True
+
+    def mark_as_done(self, cue):
+        self.ended_cues_count += 1
+        if self.ended_cues_count == len(self.targets):
+            self.ended_cues_count = 0
+            self._ended()
```

Comment:
If we don't stop tracking the cue when we "mark (it) as done", if the cue is started again, the counting will be off